apps/auth/xmpp/auth.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+from apps.auth.service import AuthService
+from superdesk import get_resource_service
+from superdesk.resource import Resource
+from apps.auth.errors import UserDisabledError, CredentialsAuthError
+from superdesk.errors import SuperdeskApiError
+from superdesk.users.errors import UserInactiveError
+from superdesk import utils
+from flask import current_app as app
+import requests
+import superdesk
+
+
+class XMPPAuthResource(Resource):
+    schema = {
+        'jid': {
+            'type': 'string',
+            'required': True
+        },
+        'transactionId': {
+            'type': 'string',
+            'required': True
+        },
+        'token': {
+            'type': 'string'
+        },
+        'user': Resource.rel('users', True)
+    }
+    resource_methods = ['POST']
+    item_methods = ['GET', 'DELETE']
+    public_methods = ['POST', 'DELETE']
+    extra_response_fields = ['user', 'token', 'username']
+
+
+superdesk.intrinsic_privilege('auth_xmpp', method=['DELETE'])
+
+
+class XMPPAuthService(AuthService):
+
+    def authenticate(self, credentials):
+        auth_url = app.config['XMPP_AUTH_URL']
+        if not auth_url:
+            raise SuperdeskApiError.notConfiguredError()
+        domain = app.config['XMPP_AUTH_DOMAIN']
+        jid = credentials.get('jid')
+        if not jid:
+            raise CredentialsAuthError(credentials)
+        user = get_resource_service('auth_users').find_one(req=None, jid=jid)
+        if not user:
+            raise CredentialsAuthError(credentials)
+
+        if 'is_enabled' in user and not user.get('is_enabled', False):
```

Comment:
after user authenticates? This means that user would enter credentials even if it can't work? Agree that this should be in BaseService as it is a common check. I've put it a AuthService method, do you think it should be somewhere else?